src/widgets/scale/lv_scale.c

```diff
@@ -182,6 +182,47 @@ void lv_scale_set_round_props(lv_obj_t * obj, uint32_t angle_range, int32_t rota
     lv_obj_invalidate(obj);
 }
 
+void lv_scale_set_line_needle_value(lv_obj_t * obj, lv_obj_t * needle_line, uint32_t needle_length, int32_t value)
+{
+    uint32_t angle;
+    lv_value_precise_t scale_width, scale_height;
+    lv_value_precise_t needle_length_x, needle_length_y;
+    float cosx_angle_value, sinx_angle_value;
```

Comment:
`float` shouldn't be used. Here a simple `int32_t` should be enough too, as we need only 1 px precision.